core/domain/summary_services.py

```diff
@@ -250,13 +257,18 @@ def get_exploration_metadata_dicts(exploration_ids, editor_user_id=None):
     for exploration_summary in exploration_summaries:
         if exploration_summary is None:
             continue
+        exploration_rights = rights_manager.get_exploration_rights(
```

Comment:
Any way we can do some kind of get_multi here? This doesn't really scale well in a for loop.

Separate PR is fine, since the current behaviour is no worse than it used to be, but please make sure to track it so this doesn't slip through the cracks.